maro/simulator/scenarios/cim/business_engine.py

```diff
@@ -615,20 +615,31 @@ def _on_action_received(self, event: CascadeEvent):
                 port_empty = port.empty
                 vessel_empty = vessel.empty
 
-                assert (
-                    -min(port.empty, vessel.remaining_space) <= move_num <= vessel_empty
-                )
+                action_type: ActionType = getattr(action, "action_type", None)
 
-                port.empty = port_empty + move_num
-                vessel.empty = vessel_empty - move_num
+                # Make it compatiable with previous action.
+                if action_type is not None:
+                    action_type = ActionType.DISCHARGE if move_num > 0 else ActionType.LOAD
 
-                event.event_type = Events.DISCHARGE_EMPTY if move_num > 0 else Events.LOAD_EMPTY
+                # Make sure the move number is positive, as we have the action type.
+                move_num = abs(move_num)
 
-                # Update cost.
-                num = abs(move_num)
+                if action_type == ActionType.DISCHARGE:
+                    assert(move_num <= vessel_empty)
```

Comment:
also need to check the port capacity?

I remember we have this attribute but just with a big enough value?

Reply:
Arthur and Ruidong removed this checking long long ago, I am not sure why, so sometime we can see that the empty+full may greater than capacity in cim vis page.